azure/services/managedclusters/spec.go

```diff
@@ -125,8 +127,21 @@ type ManagedClusterSpec struct {
 
 	// DNSPrefix allows the user to customize dns prefix.
 	DNSPrefix *string
+
 	// DisableLocalAccounts disables getting static credentials for this cluster when set. Expected to only be used for AAD clusters.
 	DisableLocalAccounts *bool
+
+	// AutoUpgradeProfile defines auto upgrade configuration.
+	AutoUpgradeProfile *ManagedClusterAutoUpgradeProfile
+}
+
+// ManagedClusterAutoUpgradeProfile auto upgrade profile for a managed cluster.
+type ManagedClusterAutoUpgradeProfile struct {
+	// NodeOSUpgradeChannel is a manner in which the OS on your nodes is updated. The default is NodeImage. Possible values include: NodeImage,Unmanaged,None
+	NodeOSUpgradeChannel *infrav1.NodeOSUpgradeChannel
```

Comment:
Note that the azure-sdk-for-go exposes a `NodeOSUpgradeChannel` property in the upgrade profile:

- https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/containerservice/armcontainerservice/models.go#L1148-L1154

The ASO version of this upgrade profile configuration does not seem to expose that:

-  https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/containerservice/armcontainerservice/models.go#L1148-L1154

Reply:
Ohhhh, so we are migrating from sdk-v4 to ASO api's ?